oracle_test.go

```diff
@@ -116,8 +116,10 @@ func TestOracle_Initialize(t *testing.T) {
 
 func TestOracle_NewUser(t *testing.T) {
 	type testCase struct {
-		creationStmts []string
-		expectErr     bool
+		creationStmts         []string
+		usernameTemplate      string
+		expectErr             bool
+		expectedUsernameRegex string
```

Comment:
I wonder if it's worth adding the `displayName` and `roleName` here so that you can check that truncation on a long `displayName` works correctly for instance (and also encapsulates the `expectedUsernameRegex` and the actual value within the same test case object).

Reply:
Good call. I've updated accordingly.